python/pyarrow/feather.py

```diff
@@ -203,15 +174,37 @@ def read_feather(source, columns=None, use_threads=True):
     columns : sequence, optional
         Only read a specific set of columns. If not provided, all columns are
         read.
+    as_pandas : bool, default True
+        Convert result to pandas.DataFrame
```

Comment:
Personally, I would not add this, as this seems inconsistent with the rest of pyarrow (or do we have this keyword elsewhere?), and the alternative is just a `.to_pandas()` call afterwards (so not even less typing). 
(and also, for pandas users, they should probably rather use `pandas.read_feather`)

Reply:
Whoops, sorry, I missed that this method actually already returns a pandas DataFrame :) (default of True).

In that case, I would propose to put the bulk of the function in `read_table` (since in principle you should be able to read a feather file into a arrow table without having pandas installed), and have `read_feather` as a small function calling `read_table(...).to_pandas(use_threads=use_threads)`

Reply:
Makes sense, will do